provisioning_client/tests/prov_transport_http_client_ut/prov_transport_http_client_ut.c

```diff
@@ -1157,7 +1177,7 @@ BEGIN_TEST_SUITE(prov_transport_http_client_ut)
         //arrange
         PROV_DEVICE_TRANSPORT_HANDLE handle = prov_dev_http_transport_create(TEST_URI_VALUE, TRANSPORT_HSM_TYPE_TPM, TEST_SCOPE_ID_VALUE, TEST_DPS_API_VALUE, on_transport_error, NULL);
```

Comment:
>TRANSPORT_HSM_TYPE_TPM [](start = 93, length = 22)

There doesn't appear to be any happy path testing for PROV_TRANSPORT_HTTP_INFO when it has an ek value. We should have a TEST_FUNCTION for that happy path because it generates a json payload specifically with those values and needs to be cleaned up.

Reply:
The test is on the http_dowork_success.  It does TPM with ek and srk

---
In reply to: [281249536](https://github.com/Azure/azure-iot-sdk-c/pull/984#discussion_r281249536) [](ancestors = 281249536)

Reply:
I am only seeing the json always being returned is TEST_JSON_CONTENT (which doesn't have any ek or srk values in it). I am pretty sure we aren't exercising the construct_json_data function and testing cleanup on line 258 of prov_transport_http_client.c..

---
In reply to: [281715420](https://github.com/Azure/azure-iot-sdk-c/pull/984#discussion_r281715420) [](ancestors = 281715420,281249536)

Reply:
All the parsing happens in the prov_client_ll.c file through the callback.  So it doesn't matter what we return here, it just get passed to the callback.

---
In reply to: [282243402](https://github.com/Azure/azure-iot-sdk-c/pull/984#discussion_r282243402) [](ancestors = 282243402,281715420,281249536)